TrackerGeom/inc/Tracker.hh

```diff
@@ -138,125 +91,25 @@ namespace mu2e {
-    TubsParams strawWallInnerMetal2(StrawId const& id) const;
-    TubsParams strawWireMother(StrawId const& id) const;
-    TubsParams strawWirePlate(StrawId const& id) const;
-  protected:
-  // G4-specific variables, set in TrackerMaker.  These should all go to G4Tracker FIXME!
-    // Position of the center of the tracker, in the Mu2e coordinate system.
-    double _z0;
-    // Outer radius of a logical volume that will just contain the entire tracker.
-    double _rOut; // is this ever used?  I think not
-
-    // these should be in a struct FIXME!
-    std::string _wallMaterialName;
-    std::string _outerMetalMaterial;
-    std::string _innerMetal1Material;
-    std::string _innerMetal2Material;
-    std::string _gasMaterialName;
-    std::string _wireMaterialName;
-    std::string _wirePlateMaterial;
-    std::string _envelopeMaterial;
-
-    // Outer envelope that holds the new style support structure.
-    PlacedTubs _mother;
-    // The envelope that holds all of the planes in the tracker,
-    // including the plane supports.
-    TubsParams _innerTrackerEnvelopeParams;
-    // The envelope that holds all of the pieces in one plane, including supports.
-    TubsParams _planeEnvelopeParams;
-    // Ditto for Panel
-    TubsParams _panelEnvelopeParams;
-    // Which level of detail is present in the model of the support structure?
-    SupportModel _supportModel;
-    // All supports are the same shape; only relevant for _supportModel=="simple"
-    Support _supportParams;
-
-    // A more detailed model of the supports; again each plane has identical supports.
-    // only relevant for _supportModel == "detailedv0".
-    SupportStructure _supportStructure;
-
-    // Deprecated: these will go away soon.
-    std::vector<double> _manifoldHalfLengths;
-    double _panelZOffset; // introduced for version 5
-
-    // Inner radius of inside edge of innermost straw.
-    double _envelopeInnerRadius;
-
-   // Electronics board
-    PanelEB _panelEB;
-
-    // non-G4 content; this is the core of the class.  These are kept when the G4 content is factorized away to a separate class FIXME!
-  private:
+    private:
     xyzVec _origin;
-    HepTransform _tracker_to_tnom; // transform to NOMINAL tracker coordinate system.  This is a null transform for the nominal tracker
     // global straw properties
     StrawProperties _strawprops;
     // Dense arrays
     PlaneCollection _planes;
     PanelCollection _panels;
+    // fundamental geometric content is in the following
+    StrawCollection _straws;
     // indirection from StrawId, for efficient lookup
     StrawIndexMap _strawindex;
     // plane existence: use cases of this should switch to using TrackerStatus and this should be removed FIXME!!
-    std::array<bool,StrawId::_nplanes> _planeExists;
-    // fundamental content is in the following
-    StrawCollection _straws;
+    PEType _planeExists;
+    // g4 content
+    std::shared_ptr<G4Tracker> _g4tracker;
```

Comment:
I am curious about the choice of shared_ptr.  I think that the Tracker class can hold this by value or by unique_ptr since it is always attached to the tracker and end users should always get it by const& or const* and never cache it.
This is not a big deal since the additional overhead is small and is incurred only at the creation time of the G4 geometry or the event display geometries.

Am I missing something?

I don't feel strongly about value vs unique_ptr.  Holding by value is, in some sense, cleaner but it would be a lot more work to change it and there *may* be some small memory optimization in keeping them separate.

Reply:
Dave convinced me that this is the right type.